client/state/index.js

```diff
@@ -149,13 +149,17 @@ export function createReduxStore( initialState = {} ) {
 
 	const middlewares = [
 		thunkMiddleware,
+		// We need the data layer middleware to be used as early
+		// as possible, before any side effects.
+		// This avoids, for example, side effect actions like analytics
+		// being dispatched more than once.
+		require( './data-layer/wpcom-api-middleware.js' ).default,
```

Comment:
it's enough of a challenge that I think another PR is fine for adding the test. I'm not sure how best to test it 🤷‍♂️